dms/security/ir.model.access.csv

```diff
@@ -3,6 +3,7 @@ id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
 access_dms_tag_user,dms_tag_user,model_dms_tag,group_dms_user,1,1,1,1
 access_dms_category_user,dms_category_user,model_dms_category,group_dms_user,1,1,1,1
 
+access_dms_storage_portal,dms_storage_portal,model_dms_storage,base.group_portal,1,0,0,0
```

Comment:
Can we avoid this permission for not exposing possible credentials stored on this model to everyone?

Reply:
No, sorry. To be able to filter record access using `ir.rule`, before you need model access from an `ir.model.access`. This line is essential for making permissions on this module behave more like Odoo permissions, which is an essential part of the patch.

However, you can see that in `security.xml` global rules are added that imply searching the corresponding computed fields, which produce the correct domain for restricting record access.

Reply:
Imagine a new module `dms_google_drive`, the usual move is to add to `dms.storage` the fields `Client secret` and `API key`. With this permission, you are giving access to any portal user to this secret information, and record rules doesn't serve, as they give whole access to the record. I think we can access to this via `sudo`, no matter other done tricks.

Reply:
Ah I see, sorry, I was confused in my previous answer and thought you were referring to access to `dms.file` or `dms.directory`. I see now that it refers to `dms.storage`.

In the case you are proposing, that hypothetical module could restrict field access [using the `groups` argument](https://www.odoo.com/documentation/12.0/developer/reference/orm.html#basic-fields).

I'll try to remove it and report anyways, as it seems a good idea in any case.

Reply:
I pushed https://github.com/OCA/dms/pull/102/commits/dd68406f7149fffcfca33c120ae01f3aa3112968 to forbid portal and basic users reading hidden `dms.storage` records. That, combined with the suggestion of filtering fields by groups, should be enough for that use case.

I've spent 2h trying to remove this line, but this is the only solution that worked without breaking all tests... Is it OK or should I spend more time on it?

Reply:
OK, just include a comment in the roadmap and it's good to go.